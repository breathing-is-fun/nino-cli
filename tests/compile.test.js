const { compile } = require('../dist/compile');
const fs = require('fs-extra');
const { joinWithRootPath } = require('../dist/utils/common');

describe('nino compile', () => {
  beforeAll(() => {
    compile({
      entry: 'tests/cases/compile',
    });
  });

  getContent = paths => fs.readFileSync(joinWithRootPath(paths)).toString();

  it('default', () => {
    const libContent = getContent(['lib', 'default.js']);
    expect(libContent).toBe(`"use strict";

var _dateFns = require("date-fns");

var defaultFunc = function defaultFunc(prefix) {
  // tslint:disable-next-line: no-console
  console.log("".concat(prefix, ": ").concat((0, _dateFns.format)(new Date(), 'yyyy-MM-dd hh:mm:ss')));
};

defaultFunc('timeStamp:');`);
  });
});
