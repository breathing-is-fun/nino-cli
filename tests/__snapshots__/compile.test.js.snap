// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nino compile compile correctly 1`] = `
"case-1
"
`;

exports[`nino compile compile correctly 2`] = `
"case-2
"
`;

exports[`nino compile compile correctly 3`] = `
"case-3
"
`;

exports[`nino compile compile correctly 4`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.calc = void 0;

var calc = function calc() {
  return 'calc';
};

exports.calc = calc;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInVua25vd24iXSwibmFtZXMiOlsiY2FsYyJdLCJtYXBwaW5ncyI6Ijs7Ozs7OztBQUFPLElBQU1BLElBQUksR0FBRyxTQUFQQSxJQUFPO0FBQUEsU0FBTSxNQUFOO0FBQUEsQ0FBYiIsInNvdXJjZXNDb250ZW50IjpbImV4cG9ydCBjb25zdCBjYWxjID0gKCkgPT4gJ2NhbGMnO1xuIl19"
`;

exports[`nino compile compile correctly 5`] = `
"case-4
"
`;

exports[`nino compile compile correctly 6`] = `
"case-5
"
`;

exports[`nino compile compile correctly 7`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.add = void 0;

require(\\"./css.css\\");

var add = function add() {
  return 2;
};

exports.add = add;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInVua25vd24iXSwibmFtZXMiOlsiYWRkIl0sIm1hcHBpbmdzIjoiOzs7Ozs7O0FBQUE7O0FBQ08sSUFBTUEsR0FBRyxHQUFHLFNBQU5BLEdBQU07QUFBQSxTQUFNLENBQU47QUFBQSxDQUFaIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0ICcuL2Nzcy5jc3MnO1xuZXhwb3J0IGNvbnN0IGFkZCA9ICgpID0+IDI7XG4iXX0="
`;

exports[`nino compile compile correctly 8`] = `
"body {
  width: 100%;
  height: 500px;
  background: #f96;
}
"
`;

exports[`nino compile compile correctly 9`] = `
"case-1
"
`;

exports[`nino compile compile correctly 10`] = `
"case-2
"
`;

exports[`nino compile compile correctly 11`] = `
"case-3-tsx
"
`;

exports[`nino compile compile correctly 12`] = `
"\\"use strict\\";

var _dateFns = require(\\"date-fns\\");

var _calcTsx = require(\\"./src/calc-tsx\\");

var defaultFunc = function defaultFunc(prefix) {
  // tslint:disable-next-line: no-console
  console.log(\\"\\".concat(prefix, \\": \\").concat((0, _dateFns.format)(new Date(), 'yyyy-MM-dd hh:mm:ss'))); // tslint:disable-next-line: no-console

  console.log((0, _calcTsx.calc)());
};

defaultFunc('timeStamp:');
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInVua25vd24iXSwibmFtZXMiOlsiZGVmYXVsdEZ1bmMiLCJwcmVmaXgiLCJjb25zb2xlIiwibG9nIiwiRGF0ZSJdLCJtYXBwaW5ncyI6Ijs7QUFBQTs7QUFDQTs7QUFDQSxJQUFNQSxXQUFXLEdBQUcsU0FBZEEsV0FBYyxDQUFDQyxNQUFELEVBQVk7QUFDNUI7QUFDQUMsRUFBQUEsT0FBTyxDQUFDQyxHQUFSLFdBQWVGLE1BQWYsZUFBMEIscUJBQU8sSUFBSUcsSUFBSixFQUFQLEVBQW1CLHFCQUFuQixDQUExQixHQUY0QixDQUc1Qjs7QUFDQUYsRUFBQUEsT0FBTyxDQUFDQyxHQUFSLENBQVksb0JBQVo7QUFDSCxDQUxEOztBQU1BSCxXQUFXLENBQUMsWUFBRCxDQUFYIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgZm9ybWF0IH0gZnJvbSAnZGF0ZS1mbnMnO1xuaW1wb3J0IHsgY2FsYyB9IGZyb20gJy4vc3JjL2NhbGMtdHN4JztcbmNvbnN0IGRlZmF1bHRGdW5jID0gKHByZWZpeCkgPT4ge1xuICAgIC8vIHRzbGludDpkaXNhYmxlLW5leHQtbGluZTogbm8tY29uc29sZVxuICAgIGNvbnNvbGUubG9nKGAke3ByZWZpeH06ICR7Zm9ybWF0KG5ldyBEYXRlKCksICd5eXl5LU1NLWRkIGhoOm1tOnNzJyl9YCk7XG4gICAgLy8gdHNsaW50OmRpc2FibGUtbmV4dC1saW5lOiBuby1jb25zb2xlXG4gICAgY29uc29sZS5sb2coY2FsYygpKTtcbn07XG5kZWZhdWx0RnVuYygndGltZVN0YW1wOicpO1xuIl19"
`;

exports[`nino compile compile correctly 13`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.calc = void 0;

var calc = function calc() {
  return 'calc';
};

exports.calc = calc;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInVua25vd24iXSwibmFtZXMiOlsiY2FsYyJdLCJtYXBwaW5ncyI6Ijs7Ozs7OztBQUFPLElBQU1BLElBQUksR0FBRyxTQUFQQSxJQUFPO0FBQUEsU0FBTSxNQUFOO0FBQUEsQ0FBYiIsInNvdXJjZXNDb250ZW50IjpbImV4cG9ydCBjb25zdCBjYWxjID0gKCkgPT4gJ2NhbGMnO1xuIl19"
`;

exports[`nino compile compile correctly 14`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.add = void 0;

require(\\"./css-tsx.css\\");

var add = function add() {
  return 2;
};

exports.add = add;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInVua25vd24iXSwibmFtZXMiOlsiYWRkIl0sIm1hcHBpbmdzIjoiOzs7Ozs7O0FBQUE7O0FBQ08sSUFBTUEsR0FBRyxHQUFHLFNBQU5BLEdBQU07QUFBQSxTQUFNLENBQU47QUFBQSxDQUFaIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0ICcuL2Nzcy10c3guY3NzJztcbmV4cG9ydCBjb25zdCBhZGQgPSAoKSA9PiAyO1xuIl19"
`;

exports[`nino compile compile correctly 15`] = `
"body {
  width: 100%;
  height: 500px;
  background: #f96;
}
"
`;

exports[`nino compile compile correctly 16`] = `
"case-4
"
`;

exports[`nino compile compile correctly 17`] = `
"case-5
"
`;

exports[`nino compile compile correctly 18`] = `
"case-1
"
`;

exports[`nino compile compile correctly 19`] = `
"case-2
"
`;

exports[`nino compile compile correctly 20`] = `
"case-3
"
`;

exports[`nino compile compile correctly 21`] = `
"export var calc = function calc() {
  return 'calc';
};
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInVua25vd24iXSwibmFtZXMiOlsiY2FsYyJdLCJtYXBwaW5ncyI6IkFBQUEsT0FBTyxJQUFNQSxJQUFJLEdBQUcsU0FBUEEsSUFBTztBQUFBLFNBQU0sTUFBTjtBQUFBLENBQWIiLCJzb3VyY2VzQ29udGVudCI6WyJleHBvcnQgY29uc3QgY2FsYyA9ICgpID0+ICdjYWxjJztcbiJdfQ=="
`;

exports[`nino compile compile correctly 22`] = `
"case-4
"
`;

exports[`nino compile compile correctly 23`] = `
"case-5
"
`;

exports[`nino compile compile correctly 24`] = `
"import './css.css';
export var add = function add() {
  return 2;
};
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInVua25vd24iXSwibmFtZXMiOlsiYWRkIl0sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLFdBQVA7QUFDQSxPQUFPLElBQU1BLEdBQUcsR0FBRyxTQUFOQSxHQUFNO0FBQUEsU0FBTSxDQUFOO0FBQUEsQ0FBWiIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCAnLi9jc3MuY3NzJztcbmV4cG9ydCBjb25zdCBhZGQgPSAoKSA9PiAyO1xuIl19"
`;

exports[`nino compile compile correctly 25`] = `
"body {
  width: 100%;
  height: 500px;
  background: #f96;
}
"
`;

exports[`nino compile compile correctly 26`] = `
"case-1
"
`;

exports[`nino compile compile correctly 27`] = `
"case-2
"
`;

exports[`nino compile compile correctly 28`] = `
"case-3-tsx
"
`;

exports[`nino compile compile correctly 29`] = `
"import { format } from 'date-fns';
import { calc } from './src/calc-tsx';

var defaultFunc = function defaultFunc(prefix) {
  // tslint:disable-next-line: no-console
  console.log(\\"\\".concat(prefix, \\": \\").concat(format(new Date(), 'yyyy-MM-dd hh:mm:ss'))); // tslint:disable-next-line: no-console

  console.log(calc());
};

defaultFunc('timeStamp:');
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInVua25vd24iXSwibmFtZXMiOlsiZm9ybWF0IiwiY2FsYyIsImRlZmF1bHRGdW5jIiwicHJlZml4IiwiY29uc29sZSIsImxvZyIsIkRhdGUiXSwibWFwcGluZ3MiOiJBQUFBLFNBQVNBLE1BQVQsUUFBdUIsVUFBdkI7QUFDQSxTQUFTQyxJQUFULFFBQXFCLGdCQUFyQjs7QUFDQSxJQUFNQyxXQUFXLEdBQUcsU0FBZEEsV0FBYyxDQUFDQyxNQUFELEVBQVk7QUFDNUI7QUFDQUMsRUFBQUEsT0FBTyxDQUFDQyxHQUFSLFdBQWVGLE1BQWYsZUFBMEJILE1BQU0sQ0FBQyxJQUFJTSxJQUFKLEVBQUQsRUFBYSxxQkFBYixDQUFoQyxHQUY0QixDQUc1Qjs7QUFDQUYsRUFBQUEsT0FBTyxDQUFDQyxHQUFSLENBQVlKLElBQUksRUFBaEI7QUFDSCxDQUxEOztBQU1BQyxXQUFXLENBQUMsWUFBRCxDQUFYIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgZm9ybWF0IH0gZnJvbSAnZGF0ZS1mbnMnO1xuaW1wb3J0IHsgY2FsYyB9IGZyb20gJy4vc3JjL2NhbGMtdHN4JztcbmNvbnN0IGRlZmF1bHRGdW5jID0gKHByZWZpeCkgPT4ge1xuICAgIC8vIHRzbGludDpkaXNhYmxlLW5leHQtbGluZTogbm8tY29uc29sZVxuICAgIGNvbnNvbGUubG9nKGAke3ByZWZpeH06ICR7Zm9ybWF0KG5ldyBEYXRlKCksICd5eXl5LU1NLWRkIGhoOm1tOnNzJyl9YCk7XG4gICAgLy8gdHNsaW50OmRpc2FibGUtbmV4dC1saW5lOiBuby1jb25zb2xlXG4gICAgY29uc29sZS5sb2coY2FsYygpKTtcbn07XG5kZWZhdWx0RnVuYygndGltZVN0YW1wOicpO1xuIl19"
`;

exports[`nino compile compile correctly 30`] = `
"export var calc = function calc() {
  return 'calc';
};
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInVua25vd24iXSwibmFtZXMiOlsiY2FsYyJdLCJtYXBwaW5ncyI6IkFBQUEsT0FBTyxJQUFNQSxJQUFJLEdBQUcsU0FBUEEsSUFBTztBQUFBLFNBQU0sTUFBTjtBQUFBLENBQWIiLCJzb3VyY2VzQ29udGVudCI6WyJleHBvcnQgY29uc3QgY2FsYyA9ICgpID0+ICdjYWxjJztcbiJdfQ=="
`;

exports[`nino compile compile correctly 31`] = `
"import './css-tsx.css';
export var add = function add() {
  return 2;
};
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInVua25vd24iXSwibmFtZXMiOlsiYWRkIl0sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLGVBQVA7QUFDQSxPQUFPLElBQU1BLEdBQUcsR0FBRyxTQUFOQSxHQUFNO0FBQUEsU0FBTSxDQUFOO0FBQUEsQ0FBWiIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCAnLi9jc3MtdHN4LmNzcyc7XG5leHBvcnQgY29uc3QgYWRkID0gKCkgPT4gMjtcbiJdfQ=="
`;

exports[`nino compile compile correctly 32`] = `
"body {
  width: 100%;
  height: 500px;
  background: #f96;
}
"
`;

exports[`nino compile compile correctly 33`] = `
"case-4
"
`;

exports[`nino compile compile correctly 34`] = `
"case-5
"
`;
